crates/iceberg/src/io.rs

```diff
@@ -368,6 +368,9 @@ impl Storage {
         new_props.insert("root".to_string(), DEFAULT_ROOT_PATH.to_string());
 
         match scheme {
+            Scheme::Memory => Ok(Self::LocalFs {
```

Comment:
To enable writing for unit tests, I add a schema for memory but consider it as LocalFs Storage, do we need to add a storage type for storage? cc @liurenjie1024 @Xuanwo

Reply:
I try to avoid adding code just for the sake of testing :)

Reply:
>  inherently you're testing a different path than it would normally would.

Yes, but we only limit it to the storage layer. It can provide a writable mock layer for unit tests which can make us test the process logic and internal state in units easier. Otherwise, we only can test this in e2e test. And the problem with e2e test is that we can't check the internal state. 🤔

Reply:
I would highly recommend the Rest catalog that we use for PyIceberg as well: https://github.com/apache/iceberg-python/blob/996afd0c44717d6ac345b8419bf01b25be2d6051/dev/docker-compose-integration.yml#L43-L56

The nice thing here that it will also validate the responses